corelib/EAST/EAST_utils.py

```diff
@@ -0,0 +1,179 @@
+
+        p_rotate = np.concatenate([p_rotate_x, p_rotate_y], axis=2)  # N*5*2
+
+        p3_in_origin = origin_1 - p_rotate[:, 4, :]
+        new_p0 = p_rotate[:, 0, :] + p3_in_origin  # N*2
+        new_p1 = p_rotate[:, 1, :] + p3_in_origin
+        new_p2 = p_rotate[:, 2, :] + p3_in_origin
+        new_p3 = p_rotate[:, 3, :] + p3_in_origin
+
+        new_p_1 = np.concatenate([new_p0[:, np.newaxis, :], new_p1[:, np.newaxis, :],
+                                  new_p2[:, np.newaxis, :], new_p3[:, np.newaxis, :]], axis=1)  # N*4*2
+    else:
+        new_p_1 = np.zeros((0, 4, 2))
+    return np.concatenate([new_p_0, new_p_1])
+
+
+def postprocess(score_map, geo_map, score_map_thresh=0.01, box_thresh=0.01, nms_thres=0.2):
+    """     Removal of boxes according to various thresholds
+    Args:
+            *   score_map: confidence score of boxes
+            *   geo_map: coordinates of boxes
+            *   score_map_thresh: Threshold to filter according to score map
+            *   box_thresh: Threshold to filter resultant boxes
+            *   nms_thres: Threshold to filter according to Non-Maximum Suppression
+    Workflow:
+            *   sort the text boxes in accordance to the y axis
+            *   restore_rectangle is called to handle tilted boxes
+            *   Locality-Aware Non-Maximum Suppression (LANMS) is used to filter
+                out overlapping boxes
+            *   Average Score Map is used to filter out more boxes
+            *   box_thresh is finally used to further filter out boxes
+    Returns:
+            *   Rezised Image along  with resizing factor
+    """
+
+    logger.info(msg="postprocess called")
+    if len(score_map.shape) == 4:
+        score_map = score_map[0, :, :, 0]
+        geo_map = geo_map[0, :, :, ]
+    xy_text = np.argwhere(score_map > score_map_thresh)
+    xy_text = xy_text[np.argsort(xy_text[:, 0])]
+    text_box_restored = restore_rectangle(xy_text[:, ::-1] * 4, geo_map[xy_text[:, 0], xy_text[:, 1], :])
+    boxes = np.zeros((text_box_restored.shape[0], 9), dtype=np.float32)
+    boxes[:, :8] = text_box_restored.reshape((-1, 8))
+    boxes[:, 8] = score_map[xy_text[:, 0], xy_text[:, 1]]
+
+    boxes = lanms.merge_quadrangle_n9(boxes.astype('float32'), nms_thres)
+
+    if boxes.shape[0] == 0:
+        return None
+
+    for i, box in enumerate(boxes):
+        mask = np.zeros_like(score_map, dtype=np.uint8)
+        cv2.fillPoly(mask, box[:8].reshape((-1, 4, 2)).astype(np.int32) // 4, 1)
+        boxes[i, 8] = cv2.mean(score_map, mask)[0]
+    boxes = boxes[boxes[:, 8] > box_thresh]
+
+    return boxes
+
+
+def sort_poly(poly):
+
+    logger.info(msg="sort_poly called")
+    min_axis = np.argmin(np.sum(poly, axis=1))
+    poly = poly[[min_axis, (min_axis + 1) % 4, (min_axis + 2) % 4, (min_axis + 3) % 4]]
+    if abs(poly[0, 0] - poly[1, 0]) > abs(poly[0, 1] - poly[1, 1]):
+        return poly
+    else:
+        return poly[[0, 3, 2, 1]]
```

Comment:
what does it mean? why is it hardcoded?

Reply:
this function decides whether to return a bounding box vertically or horizontally
p = p[0,1,2,3] when width>height
and p = p[0,3,2,1] when height>width